src/core/swap.c

```diff
@@ -139,10 +139,13 @@ static void swap_init(Unit *u) {
         assert(s);
         assert(UNIT(s)->load_state == UNIT_STUB);
 
-        s->timeout_usec = u->manager->default_timeout_start_usec;
+        if (u->manager) {
+                s->timeout_usec = u->manager->default_timeout_start_usec;
 
-        s->exec_context.std_output = u->manager->default_std_output;
-        s->exec_context.std_error = u->manager->default_std_error;
+                s->exec_context.std_output = u->manager->default_std_output;
+                s->exec_context.std_error = u->manager->default_std_error;
```

Comment:
So here's some refactoring I'd like to see done here:

Let's add a new struct UnitDefaults or so, where we place the various default_xyz options currently in Manager. We then embedd one of these structs into Manager, but we can for example define a static one inside other contexts too, such as the executor binary, with different defaults.

Morevoer, this could later be reused in main.c, where we currently have another set of arg_default_xyz variables which we so far manually copy into the Manager objects default_xyz fields.

Such a refactoring would make things a ton nicer I think, since for code pieces like here we could avoid the conditionalization on u->manager, and instead just require that the defaults object of choice is passed in unconditionally.

Reply:
The defaults are coming from the config files that the manager is loading and parsing, that's why it comes from the manager object. So it wouldn't make any difference if they are stored elsewhere, this conditionalization would be the same, because the point of it is to allow initializing the unit object from the executor binary.

Reply:
the unit object is really only used for the id and for logging, so I'll instead refactor and stop serializing it, that way most of these condizionalisations can be removed again